lib/exabgp/bgp/message/update/attribute/bgpls/link/sradj.py

```diff
@@ -56,29 +54,26 @@ def unpack (cls,data,length):
 		# *  A 4 octet index defining the offset in the SID/Label space
 		# 	 advertised by this router using the encodings defined in
 		#  	 Section 3.1.  In this case V and L flags MUST be unset.
-		sids = []
 		raw = []
 		while data:
 			# Range Size: 3 octet value indicating the number of labels in
 			# the range.
-			if int(flags.flags['V']) and int(flags.flags['L']):
+			if int(flags['V']) and int(flags['L']):
 				b = BitArray(bytes=data[:3])
-				sid = b.unpack('uintbe:24')[0:1]
+				sid = b.unpack('uintbe:24')[0]
 				data = data[3:]
-				sids.append(sid)
-			elif (not flags.flags['V']) and \
-				(not flags.flags['L']):
-				sid = unpack('!I',data[:4])[0:1]
+			elif (not flags['V']) and (not flags['L']):
+				sid = unpack('!I',data[:4])[0]
 				data = data[4:]
-				sids.append(sid)
 			else:
 				raw.append(hexstring(data))
 				break
-
-		return cls(flags=flags, sids=sids, weight=weight,undecoded=raw)
+		cls._sr_adj_sids.append({'flags': flags, 'weight': weight, 'sid': sid, 'undecoded': raw})
+		return cls()
 
 	def json (self,compact=None):
-		return ', '.join(['"sr-adj-flags": {}'.format(self.flags.json()),
-			'"sids": {}'.format(json.dumps(self.sids)),
-			'"undecoded-sids": {}'.format(json.dumps(self.undecoded)),
-			'"sr-adj-weight": {}'.format(json.dumps(self.weight))])
+		return '"sr-adj-sids": {}'.format(json.dumps(self.sr_adj_sids))
```

Comment:
I have no problem with doing this, however keep in mind it's a breaking change. Not sure how you want to handle this (eg if it requires a different major version release or similar.